fixups/EnvVarFixup/InitializeFixup.cpp

```diff
@@ -157,29 +165,57 @@ void InitializeConfiguration()
-                    auto variablevalue = specObject.get("value").as_string().wstring();
-                    traceDataStream << " value: " << variablevalue << " ;";
-
-                    auto useregistry = specObject.get("useregistry").as_string().wstring();
-                    traceDataStream << " useregistry: " << useregistry << " ;";
-                    LogString(0, "GetEnvFixup Config: name", variablenamePattern.data());
-                    LogString(0, "GetEnvFixup Config: value", variablevalue.data());
-                    LogString(0, "GetEnvFixup Config: useregistry", useregistry.data());
-                    g_envvar_envVarSpecs.emplace_back();
-                    g_envvar_envVarSpecs.back().variablename.assign(variablenamePattern.data(), variablenamePattern.length());
-                    g_envvar_envVarSpecs.back().variablevalue = variablevalue;
-                    if (useregistry.compare(L"true") == 0 ||
-                        useregistry.compare(L"True") == 0 ||
-                        useregistry.compare(L"TRUE") == 0)
+                    auto variableNamePattern = specObject.get("name").as_string().wstring();
+                    traceDataStream << " name: " << variableNamePattern << " ;";
+
+                    auto variableValue = specObject.get("value").as_string().wstring();
+                    traceDataStream << " value: " << variableValue << " ;";
+
+                    auto registry = specObject.try_get("useregistry");
+                    auto dependency = specObject.try_get("dependency");
+
+                    if (registry && dependency)
+                    {
+                        throw std::logic_error("EnvVarFixup: Bad config. Both 'useregistry' and 'dependency' specified");
+                    }
+                    else if (!registry && !dependency)
+                    {
+                        throw std::logic_error("EnvVarFixup: Bad config. Neither 'useregistry' and 'dependency' specified");
+                    }
+                    else if (registry)
                     {
-                        g_envvar_envVarSpecs.back().useregistry = true;
+                        auto useRegistry = registry->as_string().wstring();
+                        traceDataStream << " useregistry: " << useRegistry << " ;";
+                        LogString(0, "GetEnvFixup Config: name", variableNamePattern);
+                        LogString(0, "GetEnvFixup Config: value", variableValue);
+                        LogString(0, "GetEnvFixup Config: useregistry", useRegistry);
+                        g_envvar_envVarSpecs.emplace_back();
+                        g_envvar_envVarSpecs.back().remediationType = Env_Remediation_Type_Registry;
+                        g_envvar_envVarSpecs.back().variablename.assign(variableNamePattern.data(), variableNamePattern.length());
+                        g_envvar_envVarSpecs.back().variablevalue = variableValue;
+                        if (useRegistry.compare(L"true") == 0 ||

```

Comment:
If you change `userRegustry` to all lowercase you only need one comparison.

Reply:
It will also allow `tRUe`, `TrUe`,... Do we want that?